src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/eye_ref.cl

```diff
@@ -0,0 +1,21 @@
+
+#include "include/batch_headers/fetch_data.cl"
+
+KERNEL(eye_ref)(const __global INPUT0_TYPE* input, __global OUTPUT_TYPE* C) {
+    // Get the index of the current element to be processed
+    const uint united = get_global_id(0);
+    const uint batch = united / (ROW_NUMBER * COL_NUMBER);
+    const uint xy = united % (ROW_NUMBER * COL_NUMBER);
+    const uint y = xy / COL_NUMBER;
+    const uint x = xy % COL_NUMBER;
+    int idx = batch * ROW_NUMBER * COL_NUMBER + y * COL_NUMBER + x;
+
+    printf("united=%d b=%d y=%d x=%d idx=%d\n", united, batch, y, x, idx);
```

Comment:
Probably should be hidden under `ifdef` for better performance even for reference implementation

Reply:
removedd